### PR TITLE
Convert to compose bases

### LINKED ISSUE #86: Transition to compose base classes and dojo-interfaces
**Enhancement** 

Convert the existing widgets to leverage dojo-interfaces and the latest base classes from compose.

Package Version: latest

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:
- [x] There is a related issue
- [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
- [x] Unit or Functional tests are included in the PR

**Description:**

Update widgets to use `dojo-interfaces` and the `dojo-compose` `bases`

Resolves #86

### CODE REVIEW

**Comment by agubler** [comment]
Will fail until there is a release of compose

File: src/mixins/createParentListMixin.ts [hunk 1]
```diff
@@ -70,7 +71,7 @@ const createParentMixin: ParentListMixinFactory = compose<ParentList<Child>, Par
 					}
 				});
 				childrenMap.set(this, value);
-				this.emit({
+				this.emit(<any> {
```
Comment (kitsonk) [inline src/mixins/createParentListMixin.ts]: Sorry about that :-( I think dojo/interfaces#23 will fix this.

File: src/mixins/createParentListMixin.ts [hunk 2]
```diff
@@ -38,8 +39,8 @@ export interface ParentListOverloads<C extends Child> {
 	/**
 	 * Listen for events emitted from a parent when its children are mutated
 	 */
-	on(type: 'childlist', listener: EventedListener<ChildListEvent<this, C>>): Handle;
-	on(type: string, listener: EventedListener<TargettedEventObject>): Handle;
+	on?(type: 'childlist', listener: EventedListener<this, ChildListEvent<this, C>>): Handle;
```
Comment (kitsonk) [inline src/mixins/createParentListMixin.ts]: did we still need `?` with the changed interfaces?

File: src/mixins/createVNodeEvented.ts [hunk 3]
```diff
@@ -84,36 +83,29 @@ export interface VNodeEventedMixin {
...
-	on(type: 'scroll', listener: EventedListenerOrArray<UIEvent>): Handle;
-	on(type: 'submit', listener: EventedListenerOrArray<Event>): Handle;
-	/**
-	 * Add a listener to an event by type
-	 *
-	 * @param type The type of event to listen for
-	 * @param listener The event listener to attach
-	 */
-	on(type: string, listener: EventedListenerOrArray<TargettedEventObject>): Handle;
+	on(type: 'touchcancel', listener: EventedListenerOrArray<any, TouchEvent>): Handle;
```
Comment (kitsonk) [inline src/mixins/createVNodeEvented.ts]: Again, can we do `EventTarget` now?

File: tests/unit/mixins/createStatefulChildrenMixin.ts [hunk 4]
```diff
@@ -121,15 +121,15 @@ registerSuite({
 				registryProvider
 			});
 
-			parent.emit({
+			parent.emit(<any> {
```
Comment (kitsonk) [inline tests/unit/mixins/createStatefulChildrenMixin.ts]: Again, we likely can remove this with the publish of interfaces?

**Comment by agubler** [comment]
Ready for another review.

**Pull request merged**